geo/src/main/java/org/odk/collect/geo/SelectionMapFragment.kt

```diff
@@ -199,13 +216,13 @@ class SelectionMapFragment : Fragment() {
 
         summarySheetBehavior.addBottomSheetCallback(object : BottomSheetCallback() {
             override fun onStateChanged(bottomSheet: View, newState: Int) {
-                val selectedSubmissionId = selectionMapViewModel.getSelectedItemId()
+                val selectedSubmissionId = selectionViewModel.getSelectedItemId()
                 if (newState == BottomSheetBehavior.STATE_HIDDEN && selectedSubmissionId != null) {
                     map.setMarkerIcon(
                         selectedSubmissionId,
                         itemsByFeatureId[selectedSubmissionId]!!.smallIcon
```

Comment:
I can't reproduce this. It does seem like there is a problem there as you shouldn't be seeing "0 shown on map" will investigate that as well.

Reply:
Did you try again to reproduce it or it's not solved?

Reply:
I tried reproducing but couldn't cause a crash. Could you give it another go and post your exact steps if it's still happening?

Reply:
I still have that crash:

```
java.lang.NullPointerException
        at org.odk.collect.geo.SelectionMapFragment$setUpSummarySheet$1.onStateChanged(SelectionMapFragment.kt:237)
        at com.google.android.material.bottomsheet.BottomSheetBehavior.setStateInternal(BottomSheetBehavior.java:1288)
        at com.google.android.material.bottomsheet.BottomSheetBehavior$SettleRunnable.run(BottomSheetBehavior.java:1878)
        at android.view.Choreographer$CallbackRecord.run(Choreographer.java:972)
        at android.view.Choreographer.doCallbacks(Choreographer.java:796)
        at android.view.Choreographer.doFrame(Choreographer.java:727)
        at android.view.Choreographer$FrameDisplayEventReceiver.run(Choreographer.java:957)
        at android.os.Handler.handleCallback(Handler.java:938)
```

I'm not a tester so I will leave it for the QA team. Everything else seems to be fine so I would add `needs testing` label but I think you should check it more carefully because the QA team will catch it either way.

Reply:
> I still have that crash:

What steps are you going through to produce this? I've tried clicking on a marker twice, but it seems to just show and then hide the summary sheet as expected.

Reply:
1. Download the `All widgets` form
2. Record location using the `GeoWidget` an d save the form as finalized
3. Open the map of forms
4. Click on the marker once to open the summary sheet
5. Click on the marker again - **CRASH**

Reply:
Is this definitely with the latest code on this PR? Sorry I'm grasping here as it seems really strange that I can't reproduce - I'm going through those same steps (even using the same form).

Reply:
Yes I'm sure. I use an emulator Pixel 4 API 30 Google maps I don't know if it has anything to do with the issue.

Reply:
Right got it! It looks like the crash does not occur with Mapbox (I was using it to avoid needing the Google Maps API key). That's surprising but I'll investigate and see if I can work out what's going on.

Reply:
Oh wow this is a fun one! The problem actually stems from the fact that `GoogleMapFragment` uses a marker under the hood to display the crosshairs. This means that we get passed an ID we don't recognize to the `onFeatureClicked` listener. We should protect from that in `SelectionMapFragment` (it makes sense to just ignore features that aren't mappable items) but I'll also have a look and see if we want to be filtering out those "feature clicks" in `GoogleMapFragment`.